### PR TITLE
Refactor threading in DownloadManager

### PR DESCRIPTION
**Pull request by bhlieberman**

### CODE REVIEW

File: biobricks/dvc_fetcher.py [hunk 1]
```diff
@@ -158,7 +108,7 @@ def _link_cache_to_brick(self, cache_path, brick_path):
         else:
             os.symlink(cache_path, brick_path)
         
-    def fetch_outs(self, brick, prefixes=['brick/', 'data/']) -> tuple[list[dict], int]:
+    def fetch_outs(self, brick, prefixes=['brick/', 'data/', 'list/']) -> tuple[list[dict], int]:
```
Comment (zmughal) [inline biobricks/dvc_fetcher.py:111]: We shouldn't need `list/` here. The bricks should be eventually all be standardized to use `brick/` as the output.

**Comment by tomlue** [comment]
This is failing the github action test. @bhlieberman can you fix that before we pull?

**Comment by bhlieberman** [comment]
I am not sure but it seems like this test could be failing because I forked the repo and don't have read access to the necessary Github Actions secret?

**Comment by zmughal** [comment]
I pushed to a temporary branch to let it run and this is the output: <https://github.com/biobricks-ai/biobricks/actions/runs/9067245905/job/24911906739>.

**Pull request closed without merging**